config/__tests__/utils.test.ts

```diff
@@ -0,0 +1,394 @@
+    });
+  });
+
+  describe('getLocalConfigFilePath()', () => {
+    it('returns the nearest config file path', () => {
+      const mockConfigPath = '/mock/path/hubspot.config.yml';
+      mockFindup.mockReturnValue(mockConfigPath);
+
+      const localConfigPath = getLocalConfigFilePath();
+      expect(localConfigPath).toBe(mockConfigPath);
+    });
+
+    it('returns null if no config file found', () => {
+      mockFindup.mockReturnValue(null);
+      const localConfigPath = getLocalConfigFilePath();
+      expect(localConfigPath).toBeNull();
+    });
+  });
+
+  describe('getLocalConfigFileDefaultPath()', () => {
+    it('returns the default config path in current directory', () => {
+      const mockCwdPath = '/mock/cwd';
+      mockCwd.mockReturnValue(mockCwdPath);
+
+      const defaultPath = getLocalConfigFileDefaultPath();
+      expect(defaultPath).toBe(`${mockCwdPath}/hubspot.config.yml`);
+    });
+  });
+
+  describe('getConfigPathEnvironmentVariables()', () => {
+    it('returns environment config settings', () => {
+      const configPath = 'config/path';
+      process.env[ENVIRONMENT_VARIABLES.HUBSPOT_CONFIG_PATH] = configPath;
+
+      const result = getConfigPathEnvironmentVariables();
+      expect(result.useEnvironmentConfig).toBe(false);
+      expect(result.configFilePathFromEnvironment).toBe(configPath);
+    });
+
+    it('throws when both environment variables are set', () => {
+      process.env[ENVIRONMENT_VARIABLES.HUBSPOT_CONFIG_PATH] = 'path';
+      process.env[ENVIRONMENT_VARIABLES.USE_ENVIRONMENT_CONFIG] = 'true';
+
+      expect(() => getConfigPathEnvironmentVariables()).toThrow();
+    });
+  });
+
+  describe('readConfigFile()', () => {
+    it('reads and returns file contents', () => {
+      mockFs.readFileSync.mockReturnValue('config contents');
+      const result = readConfigFile('test');
+      expect(result).toBe('config contents');
+    });
+
+    it('throws FileSystemError on read failure', () => {
+      mockFs.readFileSync.mockImplementation(() => {
+        throw new Error('Read error');
+      });
+
+      expect(() => readConfigFile('test')).toThrow(FileSystemError);
+    });
+  });
+
+  describe('removeUndefinedFieldsFromConfigAccount()', () => {
+    it('removes undefined fields from account', () => {
+      const accountWithUndefinedFields = {
+        ...PAK_ACCOUNT,
+        defaultCmsPublishMode: undefined,
+        auth: {
+          ...PAK_ACCOUNT.auth,
+          tokenInfo: {
+            refreshToken: undefined,
+          },
+        },
+      };
+
+      const withFieldsRemoved = removeUndefinedFieldsFromConfigAccount(
+        accountWithUndefinedFields
+      );
+
+      expect(withFieldsRemoved).toEqual(PAK_ACCOUNT);
+    });
+  });
+
+  describe('writeConfigFile()', () => {
+    it('writes formatted config to file', () => {
+      // eslint-disable-next-line @typescript-eslint/no-empty-function
+      mockFs.ensureFileSync.mockImplementation(() => {});
```

Comment:
Not a blocker, but you could get rid of the ignores if you make these return undefined instead